sandbox/cri/config.toml

```diff
@@ -0,0 +1,189 @@
+  stream_server_address = "127.0.0.1"
+
+  # stream_server_port is the port streaming server is listening on.
+  stream_server_port = "0"
+
+  # enable_selinux indicates to enable the selinux support.
+  enable_selinux = false
+
+  # sandbox_image is the image used by sandbox container.
+  sandbox_image = "k8s.gcr.io/pause:3.1"
+
+  # stats_collect_period is the period (in seconds) of snapshots stats collection.
+  stats_collect_period = 10
+
+  # systemd_cgroup enables systemd cgroup support. This only works for runtime
+  # type "io.containerd.runtime.v1.linux".
+  # DEPRECATED: use Runtime.Options for runtime specific config for shim v2 runtimes.
+  #   For runtime "io.containerd.runc.v1", use the option `SystemdCgroup`.
+  systemd_cgroup = false
+
+  # enable_tls_streaming enables the TLS streaming support.
+  # It generates a self-sign certificate unless the following x509_key_pair_streaming are both set.
+  enable_tls_streaming = false
+
+  # "plugins.cri.x509_key_pair_streaming" contains a x509 valid key pair to stream with tls.
+  [plugins.cri.x509_key_pair_streaming]
+    # tls_cert_file is the filepath to the certificate paired with the "tls_key_file"
+    tls_cert_file = ""
+
+    # tls_key_file is the filepath to the private key paired with the "tls_cert_file"
+    tls_key_file = ""
+
+  # max_container_log_line_size is the maximum log line size in bytes for a container.
+  # Log line longer than the limit will be split into multiple lines. -1 means no
+  # limit.
+  max_container_log_line_size = 16384
+
+  # disable_cgroup indicates to disable the cgroup support.
+  # This is useful when the daemon does not have permission to access cgroup.
+  disable_cgroup = false
+
+  # disable_apparmor indicates to disable the apparmor support.
+  # This is useful when the daemon does not have permission to access apparmor.
+  disable_apparmor = false
+
+  # restrict_oom_score_adj indicates to limit the lower bound of OOMScoreAdj to
+  # the containerd's current OOMScoreAdj.
+  # This is useful when the containerd does not have permission to decrease OOMScoreAdj.
+  restrict_oom_score_adj = false
+
+  # "plugins.cri.containerd" contains config related to containerd
+  [plugins.cri.containerd]
+
+    # snapshotter is the snapshotter used by containerd.
+    snapshotter = "firecracker-naive"
+
+    # no_pivot disables pivot-root (linux only), required when running a container in a RamDisk with runc.
+    # This only works for runtime type "io.containerd.runtime.v1.linux".
+    # DEPRECATED: use Runtime.Options for runtime specific config for shim v2 runtimes.
+    #   For runtime "io.containerd.runc.v1", use the option `NoPivotRoot`.
+    no_pivot = false
+
+    # "plugins.cri.containerd.default_runtime" is the runtime to use in containerd.
+    [plugins.cri.containerd.default_runtime]
+      # runtime_type is the runtime type to use in containerd e.g. io.containerd.runtime.v1.linux
+      runtime_type = "aws.firecracker"
```

Comment:
Looks like this should be something like "io.containerd.runtime.v2.linux" rather than the name of the runtime, right?

Reply:
What this is doing is looking for the container shim. It does this by taking a look at the runtime type and creating `container-shim-aws-firecracker` name from `aws.firecracker`. If we were to use `io.containerd.runtime.v2.linux` it would look for `container-shim-v2-linux`. No idea why the `io.containerd` is removed, though.

Reply:
Are you sure that's the runtime type field and not the runtime name field?  container-shim-aws-firecracker is a runtime of type io.containerd.runtime.v2.linux.

Reply:
I tried using `io.containerd.runtime.v2.linux` and it failed trying to find `container-shim-v2-linux`. I can mess around with the configuration some more, but I know this configuration talks to our runtime.